src/applications/model/gaming-streaming-server.h

```diff
@@ -146,14 +146,14 @@ class GamingStreamingServer : public Application
   Address      m_peerAddress; //!< Remote peer address
   uint16_t     m_peerPort; //!< Remote peer port
 
-  struct TrafficStream
+  struct TrafficStream : public Object
```

Comment:
why are you extending `Object `instead of `SimpleRefCount`? Do you need something from it?

Reply:
Also, you haven't added the destructor as previously discussed: the `sendEvent `should be `Cancel`led and the two `Ptr`s set to `nullptr`

Reply:
> why are you extending `Object `instead of `SimpleRefCount`? Do you need something from it?

I can extend the `SimpleRefCount` and I changed the code acordingly

Reply:
> Also, you haven't added the destructor as previously discussed: the `sendEvent `should be `Cancel`led and the two `Ptr`s set to `nullptr`

Should I do this in the destructor method or in the `DoDispose`?
I'm doing this in the destructor method. (Actually not the same, I'm just cleaning the `m_trafficStreams` vector)

Reply:
you should do it in the destructor, as `DoDispose` is a method from `Object` and here you should really only extend `SimpleRefCount`

Reply:
Here, we are talking about the destructor of the `TrafficStream` struct, yeah?

Reply:
yes

Reply:
okay, perfect